### ISSUE TITLE
demo: using grid proxy

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana#256]
Provide complete REANA recipe for a demo workflow that needs to use grid proxy.

For example, calling `dasgoclient` for restricted CMS datasets.

(For CMS open datasets, `cernopendata-client` could in theory be used.)

See https://gitlab.cern.ch/clange/grid-proxy-test by @clelange

**Comment by tiborsimko** [comment]
Note: this is only for data access, not for computations. @alintulu might be looking into this.

**Comment by clelange** [comment]
From CMS Computing it seems that a better solution than everyone adding their certificates individually would be that REANA provides a proxy for all CMS users (checking egroup).

**Comment by roksys** [comment]
- [ ] Add proxy initialization logic to the `reana-job-controller`. See the Kerberos example https://github.com/reanahub/reana-job-controller/blob/ad7d17b4fc095f0cb53e925d568d4cb63078bfb4/reana_job_controller/kubernetes_job_manager.py#L173
https://github.com/reanahub/reana-job-controller/blob/ad7d17b4fc095f0cb53e925d568d4cb63078bfb4/reana_job_controller/kubernetes_job_manager.py#L263
- [ ] Add proxy filed to the` reana-workflow-engine-[serial,yadage,cwl]`. See the Serial example  https://github.com/reanahub/reana-workflow-engine-serial/commit/8200cf60f6f7bed5b624e4692b0456d734f00b8b#diff-6d29ceb19895382845929513c9bc9099
- [] Add docs - http://docs.reana.io/advanced-usage/access-control/kerberos

**Comment by lukasheinrich** [comment]
this is a similar discussion to the secrets / private images. Maybe a good option is that each reana user has access to 

* their own personal secrets
* a common set of secrets managed by the deployment (perhaps matched by e-group)

**Comment by tiborsimko** [comment]
Yes, the user jobs have access to secrets, but then every user would have to run kinit and voms proxy etc friends themselves... This does not seem efficient and may be perhaps too low-level for some users? 

The advantage of doing this automatically for the user via sidecar init containers is that the REANA system automatically takes care of setting everything up, including reauthentication for longer-living jobs. The user simply declares "I need Kerberos" via `kerberos: true` workflow clause, sets up his/her secret, and he/she does not have to care further and can focus on physics. (Similarly for grid proxy files.) 

Philosophically speaking, this is perhaps another example of the declarative approach (I want `kerberos: true` and don't care how it is done) vs imperative approach (I want to run `kinit` and eventually `k5reauth` after 24 hours) in people's workflows?

**Comment by lukasheinrich** [comment]
yes I'm all for a declarative approach and I agree

```
* mount secret
* kinit
```

is less declarative then

```
* kerberos: true
```

or 

```
* registry-access: atlas@gitlab-registry.cern.ch
```

or similar. I guess there is a little bit of a danger of introducing abstractions that only REANA@CERN can satisfy, but we can test this separately.

**Comment by alintulu** [comment]
closes  reanahub/reana-job-controller#240
